### PR TITLE
Implement authentication and authorization ep

### PR DESCRIPTION
**Pull request by NimeshPiyumantha**
[ITP-2128] [ITP-2129]

[ITP-2128]: https://fcodelabs.atlassian.net/browse/ITP-2128?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-2129]: https://fcodelabs.atlassian.net/browse/ITP-2129?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: front_end/lib/ui/manage_student_page/manage_student_page_view.dart [hunk 1]
```diff
@@ -298,28 +298,35 @@ class StudentMangeView extends StatelessWidget {
                                     dateController.text.trim().isEmpty) {
                                   showFieldError(
                                       'Text Field should not be empty.');
-                                } else if (!RegExp(r'^[a-zA-Z ]+$')
-                                    .hasMatch(nameController.text.trim())) {
+                                } else if (!RegExp(r'^[a-zA-Z ]+$').hasMatch(
```
Comment (KavinduDhananjaya) [inline front_end/lib/ui/manage_student_page/manage_student_page_view.dart:301]: move regex validation to seperate class

File: front_end/lib/ui/sign_in_page/sign_in_page_view.dart [hunk 2]
```diff
@@ -87,13 +100,55 @@ class SignInPageView extends StatelessWidget {
...
-                                        builder: (context) =>
-                                            AdminHomePageView()),
-                                  );
+                                onPressed: () async {
+                                  if (emailController.text.trim().isEmpty ||
+                                      passwordController.text.trim().isEmpty) {
+                                    showFieldError(
+                                        'Text Field should not be empty.');
+                                  } else if (!RegExp(
+                                          r'^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+$')
```
Comment (KavinduDhananjaya) [inline front_end/lib/ui/sign_in_page/sign_in_page_view.dart:109]: same here

Comment (NimeshPiyumantha) [reply front_end/lib/ui/sign_in_page/sign_in_page_view.dart]: done

File: front_end/lib/ui/manage_student_page/manage_student_page_view.dart (hunk 1 above)
Comment (NimeshPiyumantha) [reply front_end/lib/ui/manage_student_page/manage_student_page_view.dart]: done

**Pull request merged**